frontend/app/model/MembersOnlyContent.scala

```diff
@@ -0,0 +1,32 @@
+package model
+
+import com.gu.contentapi.client.model.{Asset, Content}
+
+case class MembersOnlyContent(content: Content, assetsMain: List[Asset], fields: Option[Map[String, String]]) {
+
+  val sortedImageSizes = assetsMain.flatMap { asset =>
+    for {
+      file <- asset.file.map(_.replace("http://static", "https://static-secure"))
+      width <- asset.typeData.get("width")
+    } yield (file, width)
+  }.sortBy(_._2.toInt)
+  val secureThumbnail = content.fields.flatMap(_.get("secureThumbnail"))
+  val mainImg = sortedImageSizes.lastOption.fold(secureThumbnail.getOrElse(""))(_._1)
+  val srcset = sortedImageSizes.map { imageDetails =>
+    val (secureFile, width) = imageDetails
+    secureFile + " " + width + "w"
+  }.mkString(", ")
```

Comment:
Yep `ResponsiveImage.scala` abstracts this stuff. I think a lot of this code can be abstracted out; might need a little bit more refactoring to `RichEvent` to get there though.